.github/workflows/main.yaml

```diff
@@ -89,3 +89,6 @@ jobs:
       run: tox -e doc
     - name: Check formatting
       run: tox -e format -- --check
+    - name: Check types
+      run: tox -e pyright
+      continue-on-error: true
```

Comment:
Suggest creating a separate job for this – it'll make the errors more visible but the status won't block merges. That'll also enable you to add a step for `mypy`!

Reply:
Done.

Reply:
It does create a bit of noise (failure emails), but ideally we won't have those failures for very long.